docs/datastore/react.md

```diff
@@ -0,0 +1,179 @@
+
+```javascript
+const { isLoading, error, data: tasks, subscribeToMore } = useQuery(TaskModel, filter);
+useEffect(() => {
+    const subscription = subscribeToMore(CRUDEvents.ADD, (newData) => {
+        if (!tasks) return [newData];
+        return [...tasks, newData];
+    });
+    return () => subscription.unsubscribe();
+}, []);
+```
+
+## useLazyQuery
+
+This hook provides a lazy `query` function.
+
+```javascript
+import { useLazyQuery } from "offix-datastore";
+
+const Tasks = () => {
+    const { isLoading, error, data: tasks, query } = useLazyQuery(TaskModel);
+
+    useEffect(() => query(), []);
+
+    if (isLoading) return <div>Loading ....</div>
+    if (error) return <div>{error.message}</div>
+
+    return (
+        <TaskFilterForm onFilterChanged={(filter) => query(filter)} />
+        {tasks.map((task) => <Task task={task} />)}
+    )
+}
+```
+
+The `query` function also returns a `subscribeToMore` function
+that you can use to subscribe to changes for that query. 
+
+## useUpdate
+
+The `useUpdate` works just like the `useSave` hook.
+The only difference is that you can also pass an update filter to the update function.
+
+```javascript
+import { useUpdate } from "offix-datastore";
+
+
+const EditTask = ({ task }) => {
+    const [taskDetails, setTaskDetails] = useState(task);
+    const { isLoading, error, update } = useUpdate(TaskModel);
+
+    async function handleSubmit(e) {
+        e.preventDefault();
+        const result = await update(taskDetails, { id: task.id });
+        setTaskDetails(result);
+    }
+
+    const handleOnChange = (e) => {
+        const { name, value } = e.target;
+        setTaskDetails((d) => ({ ...d, [name]: value }));
+    }
+
+    const { title, description, numberOfDaysLeft } = taskDetails;
+    return (
+        {(error ? <div>{error.message}</div>)}
+        <form onSubmit={handleSubmit}>
+            <input name="title" value={title} onChange={handleOnChange} />
+            <input name="description" value={description} onChange={handleOnChange} />
+            <input name="numberOfDaysLeft" type="number" value={numberOfDaysLeft} onChange={handleOnChange} />
+            <button type="submit">Submit</button>
+        </form>
+    )
+}
+```
+
+## useRemove
+
+```javascript
+import { useRemove } from "offix-datastore";
+
+const Task = ({ task }) => {
+    const { isLoading, remove } = useRemove(TaskModel);
+    
+    const deleteTask = async () => {
+        await remove(task);
+        ...
+    }
+    
+    return ...
+}
+```
+
+## useSubscription
+
+You can subscribe to specific events and receive changes data using this hook.
+Below, we listen for updates to the task and render them. The `data` returned
+by `useSubscription` is the data carried by the change event.
+
+```javascript
+const Task = ({ task }) => {
+    // Listen for updates on this task
+    const { data } = useSubscription(TaskModel, CRUDEvents.UPDATE, task);
```

Comment:
Question people might have: Can I do `CRUDEvents.UPDATE | CRUDEvents.DELETE`

Reply:
This seems like syntax from Java. I don't think it's possible in JS. To replicate this feature we can accept an array of CRUDEvents

Reply:
I meant that this is inline function that returns true. Array will work as well.